.github/workflows/podvm_smoketest.yaml

```diff
@@ -0,0 +1,157 @@
+name: Run smoke test on PodVM
+
+on:
+  pull_request:
+
+defaults:
+  run:
+    working-directory: src/cloud-api-adaptor/podvm-mkosi
+
+jobs:
+  smoke-test:
+    runs-on: 'ubuntu-22.04'
+    steps:
+      - uses: actions/checkout@v4
+
+      # Required by rootless mkosi on Ubuntu 24.04
+      # - name: Un-restrict user namespaces
+      #   run: sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
+
+      - name: Install build dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y \
+            genisoimage \
+            qemu-utils \
+            socat \
+            virt-manager
+          sudo snap install yq
+
+      - name: Read properties from versions.yaml
+        working-directory: src/cloud-api-adaptor
+        run: |
+          {
+            echo "MKOSI_VERSION=$(yq -e '.tools.mkosi' versions.yaml)"; 
+            echo "ORAS_VERSION=$(yq -e '.tools.oras' versions.yaml)";
+            echo "KATA_REF=$(yq -e '.oci.kata-containers.reference' versions.yaml)";
+            echo "KATA_REG=$(yq -e '.oci.kata-containers.registry' versions.yaml)";
+          } >> "$GITHUB_ENV"
+
+      - uses: oras-project/setup-oras@v1
+        with:
+          version: ${{ env.ORAS_VERSION }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build binaries
+        run: make binaries
+
+      - name: Disable TLS for agent-protocol-forwarder
+        run: |
+          mkdir -p ./resources/binaries-tree/etc/default
+          echo "TLS_OPTIONS=-disable-tls" > ./resources/binaries-tree/etc/default/agent-protocol-forwarder
```

Comment:
Do you think this is a candidate to also enable in the "developer mode" I'm thinking of in https://github.com/confidential-containers/cloud-api-adaptor/pull/2227
I'm trying to work out if there would be value in reusing the current podvm_mkosi workflow to help do this image build section, but we'd need to add support for that in. This isn't a blocker for this PR, just thinking about it whilst reviewing.

Reply:
yeah, although I'm not convinced we require a build-time flag for TLS. An empty tls config in daemon.json should just indicate that we're not using TLS.